### PR TITLE
DateBox: fix dateview scroll when pickerType is rollers (T921228, T934663)

### PR DESCRIPTION
**Pull request by ksercs**
…

### CODE REVIEW

File: js/ui/date_box/ui.date_view_roller.js [hunk 1]
```diff
@@ -221,30 +233,38 @@ const DateViewRoller = Scrollable.inherit({
 
         const currentSelectedIndexPosition = currentSelectedIndex * this._itemHeight();
         const dy = locationTop - currentSelectedIndexPosition;
-        const direction = dy > 0 || dy === 0 && currentSelectedIndex > 0 ? 1 : -1;
+
+        if(-0.1 <= dy && dy <= 0.1) {
```
Comment (DokaRus) [inline js/ui/date_box/ui.date_view_roller.js]: Let's avoid of using magic numbers

Comment (ksercs) [reply js/ui/date_box/ui.date_view_roller.js]: fixed

File: js/ui/date_box/ui.date_view_roller.js [hunk 2]
```diff
@@ -216,35 +228,48 @@ const DateViewRoller = Scrollable.inherit({
         return Math.max(Math.min(index, itemsCount - 1), 0);
     },
 
+    _isInNullVicinity: function(x) {
```
Comment (DokaRus) [inline js/ui/date_box/ui.date_view_roller.js]:
```suggestion
    _isInCurrentItemArea: function(delta) {
```
// `neighbourhood` seems a more reliable term in this context

Comment (ksercs) [reply js/ui/date_box/ui.date_view_roller.js]: fixed

**Review by DokaRus — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Pull request merged**